### PR TITLE
feat(donation): additional receipt email template variables

### PR DESCRIPTION
**Pull request by miguelpeixe**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Add support for the following new template variables to the RR receipt email:

- First name
- Last name
- Product name
- Billing frequency

image

This change should only impact the "Newspack" (WC) platform. Customers using Stripe should not support the new variables.

Along with the new values, the `*AMOUNT*` has changed so it uses a Woo function to determine the formatted value.

This PR also corrects the textdomain for the modified files so it uses `newspack-plugin` instead of `newspack`.

### How to test the changes in this Pull Request:

1. Checkout this branch and make sure you have Newspack as your RR platform
2. Edit the Receipt email (Newspack -> Reader Revenue -> Emails)
3. Confirm you see the new variables as above
4. Edit your content to include these variables (see example above)
5. Go through the donation flow for each type (one-time, monthly, and yearly)
6. Confirm all 3 emails include the new values
7. Edit your RR platform and set it to Stripe
8. Edit the receipt email and confirm the instructions do not include the new variables

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: includes/reader-revenue/woocommerce/class-woocommerce-connection.php [hunk 1]
```diff
@@ -1182,32 +1182,57 @@ public static function send_customizable_receipt_email( $enable, $order, $class
...
+		];
+		$product_map = [];
+		foreach ( $frequencies as $frequency => $label ) {
+			$product_id = Donations::get_donation_product( $frequency );
+			if ( $product_id ) {
+				$product_map[ $product_id ] = $label;
+			}
+		}
+
+		$item = array_values( $order->get_items() )[0];
```
Comment (leogermani) [inline includes/reader-revenue/woocommerce/class-woocommerce-connection.php:1197]: tip: `array_shift` is also an alternative here

**Review by leogermani — CHANGES_REQUESTED** [review]
First and last name worked fine, but frequency and product name did not.

image

I also noticed something that doesn't seem to be related with this PR, as I was able to reproduce on master.

When I do a new donation, with an unregistered user in a new session, I don't get the receipt. If I do a second donation in the same session, then I get it. As if the email is only sent after I'm logged in

File: includes/reader-revenue/woocommerce/class-woocommerce-connection.php (hunk 1 above)
Comment (miguelpeixe) [reply includes/reader-revenue/woocommerce/class-woocommerce-connection.php]: Updated in e2cb1073e3886930806c53d18e53c7817a7d8d6a

**Comment by miguelpeixe** [comment]
> First and last name worked fine, but frequency and product name did not.

I was not able to reproduce the issue. Can you confirm it was a donation product and under the "Newspack " RR platform?

> When I do a new donation, with an unregistered user in a new session, I don't get the receipt. If I do a second donation in the same session, then I get it. As if the email is only sent after I'm logged in

That's strange. The email is hooked to `woocommerce_email_enabled_customer_completed_order`, which is triggered by Woo. I don't think we modify the trigger behavior.

**Comment by leogermani** [comment]
Found the issue:

image

The `code` tag was added because I copied and pasted the string from the sidebar, and there they are in a code tag. 

When you double click the string on the sidebar to copy them, it selects the string without the astersks, which I then manually added in the editor.

It's an edge case, but not maybe more people can do the same and it's hard to know things are wrong. If you think there's anything easy to avoid that, we could add it now, otherwise we can leave it open for another PR.

I'm trying to think of a way to avoid this without doing too much work in the UI, If I come up with anything I let you know...

**Comment by leogermani** [comment]
> That's strange. The email is hooked to woocommerce_email_enabled_customer_completed_order, which is triggered by Woo. I don't think we modify the trigger behavior.

This no longer happened to me today

**Review by leogermani — APPROVED** [review]

**Comment by miguelpeixe** [comment]
Thanks! I think we should handle this edge case separately.

**Pull request merged**